### PR TITLE
Create / Update / Delete Groupon UP

### PR DESCRIPTION
**Pull request by buy**
## Goal

As a user I would like to create a new Groupon UP event and/or update an existing Groupon UP
## Changes
- [x] Added Parse call to update an existing Groupon UP
- [x] Added the feature to create a new UP event
- [x] Added the feature to delete an existing Groupon UP
- [x] Added `Done` button to `UIDatePickerView` for easier end of editing
#### Create

create-groupon-up
#### Update

update-groupon-up
#### Delete

delete-groupon-up

### CODE REVIEW

File: Groupon UP/Groupon UP/DealDetailsViewController.swift [hunk 1]
```diff
@@ -185,7 +184,8 @@ extension DealDetailsViewController {
     }
 
     func createUp() {
-        debugPrint("implement me: create an Up for dealId \(selectedDeal.uuid)")
+//        let upView = UPViewController()
```
Comment (roboxue) [inline Groupon UP/Groupon UP/DealDetailsViewController.swift]: Uncomment?

File: Groupon UP/Groupon UP/UPViewController.swift [hunk 2]
```diff
@@ -213,7 +254,7 @@ extension UPViewController {
             v.delegate = self
             v.addTarget(self, action: "grouponUPDateEditingDidBegin:", forControlEvents: UIControlEvents.EditingDidBegin)
 
-            if self.up.date != nil {
+            if self.up != nil && self.up.date != nil {
```
Comment (roboxue) [inline Groupon UP/Groupon UP/UPViewController.swift]: if let date = self.up.date {
do something with date
}

File: Groupon UP/Groupon UP/DealDetailsViewController.swift (hunk 1 above)
Comment (buy) [inline Groupon UP/Groupon UP/DealDetailsViewController.swift]: doesn't work yet lol

**Comment by buy** [comment]
Just added a `Done` button to `UIDatePickerView` to make life easier

**Comment by roboxue** [comment]
The correct way is use a UIToolbar and add a button... Wanna do that in this PR? Or you can ship it and leave it to me

**Comment by buy** [comment]
@roboxue we can leave it for now and revisit it once we finished all the desired features. Just added the `delete` button to the Groupon UP view

**Comment by buy** [comment]
shipping this for now

**Pull request merged**